fix(checker): false TS2440 when a file both augments a module and imports from it

## Summary
When a single source file contains a module augmentation (`declare module "base" { interface Opts {...} }`) **and** also imports a name from that module (`import { Opts } from "base"`), tsz falsely reports `TS2440: Import declaration conflicts with local declaration of 'Opts'`. tsc treats the augmentation member as a contribution to the module's `Opts`, not as a conflicting local declaration, so it accepts the code.

## Minimal repro
`base.d.ts`:
```ts
declare module "base" {
  export interface Opts { a: number; }
}
```
`aug.ts`:
```ts
import "base";
declare module "base" {
  interface Opts { b: string; }
}
import { Opts } from "base";
const o: Opts = { a: 1, b: "x" };
```
Command: `tsc/tsz --noEmit --strict --target esnext --lib esnext base.d.ts aug.ts`

- **tsc**: no errors.
- **tsz**: `aug.ts(5,10): error TS2440: Import declaration conflicts with local declaration of 'Opts'.`

## Expected
No diagnostics. The `interface Opts { b: string }` inside `declare module "base"` is a module augmentation member, not a top-level local declaration, so importing `Opts` from the same module is legal.

## Boundary / controls
- Splitting the augmentation and the `import { Opts }` into separate files (augmentation in `aug2.ts`, import in `use.ts`) produces **no error** in tsz — matching tsc. So the false positive is specifically triggered by augmentation + import of the same name co-located in one file.
- Same shape compiles cleanly in tsc regardless of file layout.

## Root-cause hypothesis + layer
Binder/checker (`WHO`/`WHERE`): the augmentation's interface member is being registered in the file's top-level symbol table as a "local declaration" rather than scoped to the augmented module's export table. The TS2440 import-vs-local conflict check then sees a phantom local `Opts` and flags the import. Augmentation members should merge into the target module's symbol, not the augmenting file's local scope.

## Test matrix
- Reported repro (interface augmentation + import).
- Renamed member (`Opts`/`Settings`) to prove non-hardcoded.
- Augmentation in a separate file (control, must stay clean).
- Augmentation with `export` keyword vs bare member.

AgentName: mod-hunter

--- a/crates/tsz-checker/src/declarations/import/declaration.rs
+++ b/crates/tsz-checker/src/declarations/import/declaration.rs
@@ -2399,6 +2399,13 @@ impl<'a> CheckerState<'a> {
                                             {
                                                 return false;
                                             }
+                                            // `declare module "..." { ... }` augmentation members
+                                            // merge into the augmented module's export table, not
+                                            // the augmenting file's scope, so they can't conflict
+                                            // with an import. Mirrors the Check 1 skip above.
+                                            if self.is_inside_module_augmentation(decl_idx) {
+                                                return false;
+                                            }
                                             // `declare global { ... }` places declarations in
                                             // the global scope; they can't conflict with an
                                             // import living in the enclosing module scope.
@@ -2666,6 +2673,9 @@ impl<'a> CheckerState<'a> {
                 }
                 let has_local_decl = other_sym.declarations.iter().any(|&decl_idx| {
                     if self.ctx.binder.node_symbols.get(&decl_idx.0) != Some(&other_sym_id) {
+                        return false;
+                    }
+                    if self.is_inside_module_augmentation(decl_idx) {
                         return false;
                     }
                     if self.is_inside_global_augmentation(decl_idx) {

--- a/crates/tsz-checker/tests/ts2440_tests.rs
+++ b/crates/tsz-checker/tests/ts2440_tests.rs
@@ -437,3 +437,107 @@ export = qux;
         get_diagnostics(source)
     );
 }
+
+// =========================================================================
+// Module augmentation members (`declare module "base" { interface X {} }`)
+// are contributions to the augmented module's export table, not local
+// declarations in the augmenting file. Importing the same name from that
+// module must NOT report TS2440. (Mirrors Check 1 / Check 2 symmetry.)
+// =========================================================================
+
+fn multi_file_codes(files: &[(&str, &str)], entry: &str) -> Vec<u32> {
+    crate::test_utils::check_multi_file(files, entry, CheckerOptions::default())
+        .into_iter()
+        .map(|d| d.code)
+        .collect()
+}
+
+#[test]
+fn test_augmentation_member_plus_import_of_same_name_no_ts2440() {
+    let base = r#"
+declare module "base" {
+  export interface Opts { a: number; }
+}
+"#;
+    let aug = r#"
+import "base";
+declare module "base" {
+  interface Opts { b: string; }
+}
+import { Opts } from "base";
+const o: Opts = { a: 1, b: "x" };
+"#;
+    let codes = multi_file_codes(&[("base.d.ts", base), ("aug.ts", aug)], "aug.ts");
+    assert!(
+        !codes.contains(&2440),
+        "Augmentation member + import of same name must NOT emit TS2440. Got: {codes:?}"
+    );
+}
+
+#[test]
+fn test_augmentation_member_plus_import_renamed_no_ts2440() {
+    // Same shape, different name — proves the fix is structural, not hardcoded.
+    let base = r#"
+declare module "base" {
+  export interface Settings { a: number; }
+}
+"#;
+    let aug = r#"
+import "base";
+declare module "base" {
+  interface Settings { b: string; }
+}
+import { Settings } from "base";
+const s: Settings = { a: 1, b: "x" };
+"#;
+    let codes = multi_file_codes(&[("base.d.ts", base), ("aug.ts", aug)], "aug.ts");
+    assert!(
+        !codes.contains(&2440),
+        "Renamed augmentation member + import must NOT emit TS2440. Got: {codes:?}"
+    );
+}
+
+#[test]
+fn test_augmentation_member_exported_plus_import_no_ts2440() {
+    // Augmentation member written with the `export` keyword (vs bare member).
+    let base = r#"
+declare module "base" {
+  export interface Opts { a: number; }
+}
+"#;
+    let aug = r#"
+import "base";
+declare module "base" {
+  export interface Opts { b: string; }
+}
+import { Opts } from "base";
+const o: Opts = { a: 1, b: "x" };
+"#;
+    let codes = multi_file_codes(&[("base.d.ts", base), ("aug.ts", aug)], "aug.ts");
+    assert!(
+        !codes.contains(&2440),
+        "Exported augmentation member + import must NOT emit TS2440. Got: {codes:?}"
+    );
+}
+
+#[test]
+fn test_genuine_local_interface_plus_import_still_conflicts() {
+    // Negative control: a genuine top-level local interface (NOT inside an
+    // augmentation) colliding with a type-bearing import must still report
+    // TS2440. This proves the fix only relaxes augmentation-scoped members.
+    let base = r#"
+declare module "base" {
+  export interface Opts { a: number; }
+}
+"#;
+    let use_ts = r#"
+import { Opts } from "base";
+interface Opts { b: string; }
+const o: Opts = { a: 1, b: "x" };
+"#;
+    let codes = multi_file_codes(&[("base.d.ts", base), ("use.ts", use_ts)], "use.ts");
+    assert!(
+        codes.contains(&2440),
+        "Genuine local interface colliding with import must still emit TS2440. Got: {codes:?}"
+    );
+}

